### PR TITLE
create questions with survey 

### PR DESCRIPTION
**Pull request by bkzhang**
Moved controllers to controller dir. Added the ability to add a text question, todo for the questions, add a ui to display survey + appropriate test modification to show that question appears in survey after being added.

### CODE REVIEW

**Comment by bkzhang** [comment]
Good point, since the frontend still has the add question, i reorganized it to have an `add_question` functionality but also added in the ability to create a new survey along with questions by passing in a json Survey object. Heads up @ismail-syed this will break #48 unless this PR changes

**Comment by bkzhang** [comment]
> Why have a separate controller for questions? It doesn't seem like we'd be adding questions separately from the creation of a survey. It seems like what we actually want to create the questions during the creation of the survey.
> 
> What this will mean is that we'll have to pass the questions as params to the `/new` endpoint and add them to the `Survey` before saving it.

it was only there because we had an add questions button, since its removed i can remove it, as for the other stuff you mentioned its already in the changes

File: src/main/java/com/example/survey/controllers/SurveyController.java [hunk 1]
```diff
@@ -15,32 +20,31 @@ public SurveyController(SurveyRepo surveyRepo) {
...
     }
 
-    @PostMapping("/new")
+    @PostMapping(value = "/new", produces = "application/json; charset=utf-8", consumes = MediaType.APPLICATION_JSON_VALUE)
     @ResponseBody
-    public Survey newSurvey(@RequestParam("name") String name) {
-        Survey s = new Survey(name);
-        this.surveyRepo.save(s);
-        return s;
+    public Survey newSurvey(@RequestBody Survey survey) {
```
Comment (JackyChiu) [inline src/main/java/com/example/survey/controllers/SurveyController.java]: Does this mean the question will come in the form of the following and be decoding into the concrete type? 
```
"questions": [
   { "name": "joe", "type": "text", ... }
]
```

Comment (bkzhang) [reply src/main/java/com/example/survey/controllers/SurveyController.java]: yeah! for the FE, will have to have an add question box on top of the name input, and maybe a radio list for the type of questions

**Comment by bkzhang** [comment]
So in this PR: changed the controller so that the `/new` endpoint takes in the entire survey object.

I think #49  #50  #52 should be handled in a separate pr.

**Comment by vikrambombhi** [comment]
Can you confirm this work without the `id` key. In ur example `id` is provided but the client shouldnt have to deal gerating ids for models

**Comment by bkzhang** [comment]
> Can you confirm this work without the `id` key. In ur example `id` is provided but the client shouldnt have to deal gerating ids for models

you don't have to fill out everything, it works locally rn and the only thing we have is name

File: client/src/components/NewSurveyModal.js [hunk 2]
```diff
@@ -14,8 +14,15 @@ export function NewSurveyModal({ open, openHandler }) {
...
-      const res = await fetch(`/api/new?name=${name}`, {
-        method: "POST"
+      const res = await fetch(`/api/new`, {
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            name: `${name}`,
```
Comment (ismail-syed) [inline client/src/components/NewSurveyModal.js]: Does this work now?

Comment (bkzhang) [reply client/src/components/NewSurveyModal.js]: yes

**Pull request merged**